### PR TITLE
start a command for running the template service broker

### PR DESCRIPTION
**Pull request by deads2k**
This stubs in a command with the full set of "normal" flags.  I didn't see admission plugins, so those aren't wired up.

You'll need to finish the client config wiring for clients and informers, but this shows you'll need to add those.

@liggitt @bparees @jim-minter

### CODE REVIEW

File: pkg/openservicebroker/cmd/server/start.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+	config, err := o.Config()
+	if err != nil {
+		return err
+	}
+
+	server, err := config.Complete().New(genericapiserver.EmptyDelegate)
+	if err != nil {
+		return err
+	}
+	return server.GenericAPIServer.PrepareRun().Run(stopCh)
```
Comment (juanvallejo) [inline pkg/openservicebroker/cmd/server/start.go]: nit, but maybe name `server` here something slightly different so it doesn't conflict with the imported pkg `"github.com/openshift/origin/pkg/openservicebroker/server"`

Comment (deads2k) [inline pkg/openservicebroker/cmd/server/start.go]: > nit, but maybe name server here something slightly different so it doesn't conflict with the imported pkg "github.com/openshift/origin/pkg/openservicebroker/server"

As I make it more apiserver like, I think we'll change in the other direction.

**Comment by deads2k** [comment]
We seem to be having some weird yum and docker issues across multiple pulls.

**Comment by deads2k** [comment]
https://github.com/openshift/origin/issues/8571

/test end_to_end

**Comment by deads2k** [comment]
@bparees I think you're looking for `do-not-merge`.  Comment as to why?

**Comment by deads2k** [comment]
> @bparees I think you're looking for do-not-merge. Comment as to why?

This is a building block for https://github.com/openshift/origin/pull/15672 which makes the TSB a completely separable component, including a template to use it.

File: pkg/openservicebroker/cmd/server/start.go [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+	return nil
+}
+
+func (o *TemplateServiceBrokerServerOptions) Complete() error {
+	return nil
+}
+
+func (o TemplateServiceBrokerServerOptions) Config() (*server.TemplateServiceBrokerConfig, error) {
+	// TODO have a "real" external address
+	if err := o.RecommendedOptions.SecureServing.MaybeDefaultWithSelfSignedCerts("localhost", nil, []net.IP{net.ParseIP("127.0.0.1")}); err != nil {
```
Comment (bparees) [inline pkg/openservicebroker/cmd/server/start.go]: at what point is this going to serve with a cert signed by the cluster CA?  Isn't that going to be necessary for the SC to cleanly talk to it?

Comment (deads2k) [inline pkg/openservicebroker/cmd/server/start.go]: > at what point is this going to serve with a cert signed by the cluster CA? Isn't that going to be necessary for the SC to cleanly talk to it?

Never.  Serving will happen with a cert created and signed by the service serving cert signing.  This sets up a default if you don't pass one in, making the "no args run" work.

**Pull request merged**